service/s3/unmarshal.go

```diff
@@ -0,0 +1,111 @@
+package s3
+
+import (
+	"bytes"
+	"encoding/xml"
+	"io"
+	"net/http"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/aws/awserr"
+	restlegacy "github.com/aws/aws-sdk-go-v2/private/protocol/rest"
+)
+
+// protoGetObjectUnmarshaler defines unmarshaler for ProtoGetObject operation
+type protoGetObjectUnmarshaler struct {
+	output *GetObjectOutput
+}
+
+// unmarshalOperation is the top level method used with a handler stack to unmarshal an operation response
+// This method calls appropriate unmarshal shape functions as per the output shape and protocol used by the
+// service.
+func (u protoGetObjectUnmarshaler) unmarshalOperation(r *aws.Request) {
+	var buff bytes.Buffer
+	body := io.TeeReader(r.HTTPResponse.Body, &buff)
+	// an xml decoder should be used to decode an xml document
+	d := xml.NewDecoder(body)
+	startToken, err := d.Token()
+	if err != nil && err != io.EOF {
+		r.Error = err
+		return
+	}
+
+	// isRequestError checks if operation response returned a error response
+	if isRequestError(r) {
+		r.Error = unmarshalErrorPrototype(r, d, startToken)
+		return
+	}
+
+	// unmarshalGetObjectOutputShapeBodyAWSREST unmarshal's Rest Components of response
+	if err := unmarshalGetObjectOutputShapeBodyAWSREST(u.output, r); err != nil {
+		r.Error = err
+		return
+	}
+
+	// unmarshalGetObjectOutputShapeAWSPayload unmarshal's payload for GetObject output shape
+	if err := unmarshalGetObjectOutputShapeAWSPayload(u.output, r); err != nil {
+		r.Error = err
+		return
+	}
+}
+
+// isRequestError would check if a request response was an error
+// It takes in *aws.Request, & a startTag to check for error.
+func isRequestError(r *aws.Request) bool {
+	if r.HTTPResponse.StatusCode != http.StatusOK {
```

Comment:
Probably don't want to check only for 200, there are other 200 status codes that could be returned like 204. For example the SDK's response validation handlers, `ValidateResponseHandler` checks of the status code is `0`, or 
>= `300` to determine if the response status code as an error.